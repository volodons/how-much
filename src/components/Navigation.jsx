import { useState } from "react";
import { Link } from "react-router-dom";
import MenuIcon from "@mui/icons-material/Menu";
import PriceChangeIcon from "@mui/icons-material/PriceChange";
import CurrencyExchangeIcon from "@mui/icons-material/CurrencyExchange";
import CloseIcon from "@mui/icons-material/Close";
import Box from "@mui/material/Box";
import Button from "@mui/material/Button";
import Drawer from "@mui/material/Drawer";
import List from "@mui/material/List";
import ListItem from "@mui/material/ListItem";
import ListItemButton from "@mui/material/ListItemButton";
import ListItemIcon from "@mui/material/ListItemIcon";
import ListItemText from "@mui/material/ListItemText";
import { styled } from "@mui/material/styles";

import { COLORS } from "../const/styles";
import { MEDIA_QUERIES } from "../const/styles";

const StyledOpenButton = styled(Button)`
    position: absolute;
    margin-top: 1.5rem;
    color: ${COLORS.WHITE};
    @media (max-width: ${MEDIA_QUERIES.XS}) {
        position: static;
        margin-top: 0;
    }
`;

const StyledOpenButtonIcon = styled(MenuIcon)`
    font-size: 2rem;
`;

const StyledCloseButton = styled(Button)`
    color: ${COLORS.WHITE};
`;

const StyledCloseButtonIcon = styled(CloseIcon)`
    font-size: 2rem;
    color: ${COLORS.WHITE};
`;

const StyledDrawer = styled(Drawer)`
    & .MuiDrawer-paper {
        padding: 1rem;
        background-color: ${COLORS.BLUE};
    }
`;

const StyledButtonBox = styled(Box)`
    display: flex;
    justify-content: center;
    margin-bottom: 1rem;
`;

const StyledLink = styled(Link)`
    all: unset;
`;

const StyledListBox = styled(Box)`
    color: ${COLORS.WHITE};
`;

const StyledListItemButton = styled(ListItemButton)`
    border: 2px solid ${COLORS.WHITE};
    border-radius: 10px;
    width: 280px;
`;

const StyledListItemText = styled(ListItemText)`
    & .MuiListItemText-primary {
        font-weight: bold;
    }
`;

const StyledCurrencyExchangeIcon = styled(CurrencyExchangeIcon)`
    font-size: 2rem;
    color: ${COLORS.WHITE};
`;

const StyledPriceChangeIcon = styled(PriceChangeIcon)`
    font-size: 2rem;
    color: ${COLORS.WHITE};
`;

const Navigation = () => {
    const [openDrawer, setOpenDrawer] = useState(false);

    const handleToggleDrawer = () => {
        setOpenDrawer(!openDrawer);
    };

    return (
        <>
            <StyledDrawer open={openDrawer}>
                <StyledButtonBox>
                    <StyledCloseButton onClick={handleToggleDrawer}>
                        <StyledCloseButtonIcon />
                    </StyledCloseButton>
                </StyledButtonBox>
                <StyledListBox>
                    <List>
                        <ListItem>
                            <StyledLink to="/">
                                <StyledListItemButton>
                                    <ListItemIcon>
                                        <StyledCurrencyExchangeIcon />
                                    </ListItemIcon>
                                    <StyledListItemText>
                                        Currency Converter
                                    </StyledListItemText>
                                </StyledListItemButton>
                            </StyledLink>
                        </ListItem>
                        <ListItem>
                            <StyledLink to="/rates">
                                <StyledListItemButton>
                                    <ListItemIcon>
                                        <StyledPriceChangeIcon />
                                    </ListItemIcon>
                                    <StyledListItemText>
                                        Exchange Rates
                                    </StyledListItemText>
                                </StyledListItemButton>
                            </StyledLink>
                        </ListItem>
                    </List>
                </StyledListBox>
            </StyledDrawer>
            <StyledOpenButton onClick={handleToggleDrawer}>
                <StyledOpenButtonIcon />
            </StyledOpenButton>
        </>
    );
};

export default Navigation;
