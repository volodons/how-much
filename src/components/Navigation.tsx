import React, { useState } from 'react';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';
import ListItemIcon from '@mui/material/ListItemIcon';

import {
    StyledOpenButton,
    StyledOpenButtonIcon,
    StyledCloseButton,
    StyledCloseButtonIcon,
    StyledDrawer,
    StyledButtonBox,
    StyledLink,
    StyledListBox,
    StyledListItemButton,
    StyledListItemText,
    StyledCurrencyExchangeIcon,
    StyledPriceChangeIcon,
} from '../styled/styledNavigation';

const Navigation: React.FC = () => {
    const [openDrawer, setOpenDrawer] = useState<boolean>(false);

    const handleToggleDrawer = () => {
        setOpenDrawer(!openDrawer);
    };

    return (
        <>
            <StyledDrawer open={openDrawer}>
                <StyledButtonBox>
                    <StyledCloseButton onClick={handleToggleDrawer}>
                        <StyledCloseButtonIcon />
                    </StyledCloseButton>
                </StyledButtonBox>
                <StyledListBox>
                    <List>
                        <ListItem>
                            <StyledLink to="/">
                                <StyledListItemButton>
                                    <ListItemIcon>
                                        <StyledCurrencyExchangeIcon />
                                    </ListItemIcon>
                                    <StyledListItemText>
                                        Currency Converter
                                    </StyledListItemText>
                                </StyledListItemButton>
                            </StyledLink>
                        </ListItem>
                        <ListItem>
                            <StyledLink to="/rates">
                                <StyledListItemButton>
                                    <ListItemIcon>
                                        <StyledPriceChangeIcon />
                                    </ListItemIcon>
                                    <StyledListItemText>
                                        Exchange Rates
                                    </StyledListItemText>
                                </StyledListItemButton>
                            </StyledLink>
                        </ListItem>
                    </List>
                </StyledListBox>
            </StyledDrawer>
            <StyledOpenButton onClick={handleToggleDrawer}>
                <StyledOpenButtonIcon />
            </StyledOpenButton>
        </>
    );
};

export default Navigation;
