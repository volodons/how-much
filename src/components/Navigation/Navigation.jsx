import { useState } from "react";
import Drawer from "@mui/material/Drawer";
import Button from "@mui/material/Button";
import {
    Box,
    List,
    ListItem,
    ListItemButton,
    ListItemIcon,
    ListItemText,
} from "@mui/material";
import { styled } from "@mui/material";
import MenuIcon from "@mui/icons-material/Menu";
import PriceChangeIcon from "@mui/icons-material/PriceChange";
import CurrencyExchangeIcon from "@mui/icons-material/CurrencyExchange";
import CloseIcon from "@mui/icons-material/Close";

const StyledOpenButton = styled(Button)`
    position: absolute;
    color: #f5f7ff;
`;

const StyledCloseButton = styled(Button)`
    color: #f5f7ff;
`;

const StyledButtonBox = styled(Box)`
    display: flex;
    justify-content: center;
    margin-bottom: 1rem;
`;

const StyledListBox = styled(Box)`
    color: #f5f7ff;
`;

const StyledListItemButton = styled(ListItemButton)`
    border: 2px solid #f5f7ff;
    border-radius: 10px;
`;

const StyledListItemText = styled(ListItemText)`
    & .MuiListItemText-primary {
        font-weight: bold;
    }
`;

const Navigation = () => {
    const [openDrawer, setOpenDrawer] = useState(null);

    const handleToggleDrawer = () => {
        setOpenDrawer(!openDrawer);
    };

    return (
        <>
            <Drawer
                open={openDrawer}
                sx={{
                    "& .MuiDrawer-paper": {
                        padding: "1rem",
                        backgroundColor: "#0390fc",
                    },
                }}
            >
                <StyledButtonBox>
                    <StyledCloseButton onClick={handleToggleDrawer}>
                        <CloseIcon
                            sx={{ fontSize: "2rem", color: "#f5f7ff" }}
                        />
                    </StyledCloseButton>
                </StyledButtonBox>
                <StyledListBox>
                    <List>
                        <ListItem>
                            <StyledListItemButton>
                                <ListItemIcon>
                                    <CurrencyExchangeIcon
                                        sx={{
                                            fontSize: "2rem",
                                            color: "#f5f7ff",
                                        }}
                                    />
                                </ListItemIcon>
                                <StyledListItemText>
                                    Currency Converter
                                </StyledListItemText>
                            </StyledListItemButton>
                        </ListItem>
                        <ListItem>
                            <StyledListItemButton>
                                <ListItemIcon>
                                    <PriceChangeIcon
                                        sx={{
                                            fontSize: "2rem",
                                            color: "#f5f7ff",
                                        }}
                                    />
                                </ListItemIcon>
                                <StyledListItemText>
                                    Exchange Rates
                                </StyledListItemText>
                            </StyledListItemButton>
                        </ListItem>
                    </List>
                </StyledListBox>
            </Drawer>
            <StyledOpenButton onClick={handleToggleDrawer}>
                <MenuIcon
                    sx={{
                        marginTop: "1.5rem",
                        fontSize: "2rem",
                        color: "#f5f7ff",
                    }}
                />
            </StyledOpenButton>
        </>
    );
};

export default Navigation;
