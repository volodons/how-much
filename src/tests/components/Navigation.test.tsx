import { render, screen, fireEvent } from '@testing-library/react';

import Navigation from '../../components/Navigation';

describe('Navigation component', () => {
    it('should render without errors', () => {
        render(<Navigation />);
        const openButton = screen.getByTestId('open-button');
        expect(openButton).toBeInTheDocument();
    });

    it('should open and close the drawer when open button is clicked', () => {
        render(<Navigation />);
        const openButton = screen.getByTestId('open-button');
        act(() => {
            fireEvent.click(openButton);
        });
        const drawer = screen.getByTestId('drawer');
        expect(drawer).toHaveClass(
            'MuiDrawer-root',
            'MuiDrawer-modal',
            'MuiDrawer-paper'
        );
        fireEvent.click(openButton);
        expect(drawer).not.toBeVisible();
    });

    it('should display navigation links', () => {
        render(<Navigation />);
        const currencyConverterLink = screen.getByTestId(
            'currency-converter-link'
        );
        const exchangeRatesLink = screen.getByTestId('exchange-rates-click');
        expect(currencyConverterLink).toBeInTheDocument();
        expect(exchangeRatesLink).toBeInTheDocument();
    });

    it('should navigate to correct routes when links are clicked', () => {
        render(<Navigation />);
        const currencyConverterLink = screen.getByText('Currency Converter');
        const exchangeRatesLink = screen.getByText('Exchange Rates');
        fireEvent.click(currencyConverterLink);
        fireEvent.click(exchangeRatesLink);
    });
});
