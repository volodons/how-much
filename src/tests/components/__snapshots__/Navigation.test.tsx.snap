// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component should match a snapshot 1`] = `
<div
  className="MuiBox-root css-0"
  data-testid="navigation-component"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1sg6drv-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-188uhum-MuiSvgIcon-root"
      data-testid="open-button"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
      />
    </svg>
  </button>
</div>
`;
